woocommerce/wc-admin
# Expand / collapse list items component

Putting this up to show working so far. This works in conjunction with the ExperimentalList/ ExperimentalListItem to produce a expand/collapse. This is roughly functional but the transition work still hasn't been started, the idea I had there was for this component to wrap the list item children in a new CSS transition group that supports the animation defined in the designs. 

The API works like this currently:

```js
<ExperimentalCollapsibleList
  hideText="Show less"
  showText="Show 1 more item"
  minChildrenToShow={2}
>
  <ExperimentalListItem>
	  <div>Hello world</div>
  </ExperimentalListItem>
  <ExperimentalListItem>
	  <div>Hello world</div>
  </ExperimentalListItem>
  <ExperimentalListItem>
     <div>Hello world, I am hiding</div>
  </ExperimentalListItem>
</ExperimentalCollapsibleList>
```

I contemplated adding default translated strings for the text, but since we don't have a direct need for that I suggest we just pass it in for now.

Also note that we should move this to `@woocommerce/experimental` along with the `ExperimentalList` 

No changelog, as it is in the package changelog.

### Screenshot

![list-collapse](https://user-images.githubusercontent.com/2240960/116405628-3ce76580-a806-11eb-8d44-be0d54bbb4bf.gif)

### Testing instructions

- Load this PR and start the story book `npm run storybook`
- Go to **List > ExperimentalList with ExperimentalCollapsibleListItem**
- Expand and collapse the list, it should similar as shown in https://github.com/woocommerce/woocommerce-admin/issues/6636
